ops/deploy-ccs.groovy

```diff
@@ -120,9 +186,53 @@ def buildAppAmis(AmiIds amiIds, AppBuildResults appBuildResults) {
  * @param appBuildResults (not used in the CCS environment; this stuff is all baked into the AMIs there, instead)
  * @throws RuntimeException An exception will be bubbled up if the deploy tooling returns a non-zero exit code.
  */
-def deploy(String envId, AmiIds amiIds, AppBuildResults appBuildResults) {
-	echo 'Deploy to the CCS environment is not yet implemented.'
-	throw new UnsupportedOperationException('Deploy to the CCS environment is not yet implemented.')
+def deploy(String environmentId, AmiIds amiIds, AppBuildResults appBuildResults) {
+	def env = normalizeEnvironmentId(environmentId)
+		dir("${workspace}/ops/terraform/env/${env}/stateless") {
+	
+		// Debug output terraform version 
+		sh "/usr/bin/terraform --version"
+		
+		// Initilize terraform 
+		sh "/usr/bin/terraform init"
+		
+		// Gathering terraform plan 
+		sh "/usr/bin/terraform plan \
+		-var='fhir_ami=${amiIds.bfdServerAmiId}' \
+		-var='etl_ami=${amiIds.bfdPipelineAmiId}' \
+		-var='ssh_key_name=bfd-${env}' \
+		-no-color"
+		
+		// Apply Terraform plan
+		sh "/usr/bin/terraform apply \
+		-var='fhir_ami=${amiIds.bfdServerAmiId}' \
+		-var='etl_ami=${amiIds.bfdPipelineAmiId}' \
+		-var='ssh_key_name=bfd-${env}' \
+		-no-color -auto-approve"
+	}
+}
+
+
+def extractAmiIdFromPackerManifest(File manifest) {
+	dir('ops/ansible/playbooks-ccs'){
+		def manifestJson = new JsonSlurper().parseText(manifest.text)
+
+   	// artifactId will be of the form $region:$amiId
+   	return manifestJson.builds[0].artifact_id.split(":")[1]
+	}
+}
+
+def normalizeEnvironmentId(String environmentId) {
+	switch (environmentId) {
+		case 'test':
+		case 'prod':
+		case 'prod-sbx':
+		    return environmentId
+		case 'prod-stg':
+		    return 'prod-sbx'
+		default:
+		    throw new UnsupportedOperationException("Deploy to the CCS ${environmentId} environment is not yet implemented.")
```

Comment:
Love the error-throwing default case. Thanks!

Reply:
test